src/DeploymentTemplate.ts

```diff
@@ -58,9 +59,8 @@ export class DeploymentTemplate {
      * @param _documentId A unique identifier for this document. Usually this will be a URI to the document.
      */
     constructor(private _documentText: string, private _documentId: string) {
-        assert(_documentText !== null);
-        assert(_documentText !== undefined);
-        assert(_documentId);
```

Comment:
>_documentId [](start = 17, length = 11)

How come we don't need to check _documentId anymore?

Reply:
Another good catch, surprised the suites didn't catch that.  Meant this:
```
nonNullOrEmptyValue(_documentId, "_documentId");
```
We don't actually need most of the null/undefined checks - they're left over from when the compiler couldn't check for null/undefined. I've just decided to leave them in for now, but I guess I'm not really consistent with it.